src/leveled_codec.erl

```diff
@@ -112,10 +112,17 @@
         {index_specs(), infinity|integer()}. % {KeyChanges, TTL}
 -type maybe_lookup() ::
         lookup|no_lookup.
+-type actual_regex() ::
+        reference()|{re_pattern, term(), term(), term(), term()}.
```

Comment:
is there a reason not to use the opaque type `re:mp()` instead of `re_patten...`?

Reply:
[Not exported until > OTP26](https://github.com/erlang/otp/commit/eb4a9c1e45bccf8d5463ec69198db41149eda823)